### ISSUE TITLE
cosmos.aio.create_container_if_not_exists does not call response_hook

### ISSUE THREAD

**Issue by metamoof** [Azure/azure-sdk-for-python#34281]
- **Package Name**: azure-cosmos
- **Package Version**: 4.5.1
- **Operating System**: windows
- **Python Version**: 3.10.11

**Describe the bug**
In the documentation for `azure.cosmos.aio.DatabaseProxy.create_container_if_not_exists` it states that you can pass a `response_hook` keyword that will be invoked with the response metadata. 

If the container exists, then it will be invoked.

If the container does not exist, then it will not be invoked.

**To Reproduce**
1. Create a new database
2. Connect to the database using `azure.cosmos.aio.DatabaseProxy`
3. Execute `await database.create_container_if_not_exists(id='foo', partitionkey = azure.cosmos.PartitionKey(path="/id"), offer_throughput='1000', response_hook=hook)` where `hook` is a callable
4. `hook` will not be called. 

**Additional context**
The problem seems to be in the call to `self.create_container` in [sdk/cosmos/aio/_database.py:318](https://github.com/Azure/azure-sdk-for-python/blob/506bffb753f7363cb72337aca46aedac654e4628/sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py#L318) that is not including the keyword `response_hook`

**Comment by pvaneck** [comment]
Thanks for the issue report/feedback, we'll investigate as soon as possible.

**Comment by annatisch** [comment]
@simorenoh - this seems to be indicative of a broader problem in that we are not passing `**kwargs` through to the create_container method - which would also imact any other pipeline configuration kwargs alongside response_hook:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/azure/cosmos/database.py#L320-L331

The response_hook support in create_container itself _should_ now be fixed (still awaiting release) due to the refactor of moving this down to the connection layer - though it's still missing documentation.

**Comment by simorenoh** [comment]
@annatisch oh god I just saw that this is for both the sync and async ones as well - PR to fix all of it here: https://github.com/Azure/azure-sdk-for-python/pull/34286 - although not sure what you mean about the documentation since I do see the `response_hook` kwarg description in all the methods involved (create_if_not_exists, create_container, read (on container proxy)).

thanks @metamoof for finding this, and sorry about the gap in expected behavior - will make sure to update this issue once we release the changes.

**Comment by annatisch** [comment]
Ah you're right - it totally is documented. Ignore me :D

**Comment by metamoof** [comment]
Thank you for fixing it so quickly.

Whilst this particular thing is documented, the only reason I'm looking at it is because I cannot see documentation on whether it is needed or not. If I just `await create_container_if_not_exists()` is that guaranteed to not return until the container is created or an exception is thrown? If so, then I don't need to do the really awkward thing I'm doing with awaiting `future` objects that are being triggered by the callbacks.

Some clarity on what is expected when we await these actions, whether it's a promise or an action, would be helpful.

**Comment by simorenoh** [comment]
@metamoof Nope, the kwarg is not needed - only if you'd like to directly link a Callable to execute on the results. Kwargs are always strictly optional, which is why they are not present in the direct method definition.

We have several samples within the SDK that you can use to see how to use all of our methods - I recommend you go through them since you will likely find most things you're looking for: https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/cosmos/azure-cosmos/samples

Here are the ones for async container operations for instance: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/samples/container_management_async.py
